### PR TITLE
check if OpenCV_LIB_COMPONENTS contains opencv_contrib

### PR DESCRIPTION
**Pull request by k-okada**
see #108 for discussion

### CODE REVIEW

File: opencv_apps/CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,10 @@ catkin_package()
 include_directories( ${catkin_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS})
 
 ## Declare a cpp library
+if("${OpenCV_LIB_COMPONENTS}" MATCHES ";opencv_contrib;")
```
Comment (vrabaud) [inline opencv_apps/CMakeLists.txt]: I'd rather keep it specific  `if("${OpenCV_LIB_COMPONENTS}" MATCHES "opencv_optflow")` no ?

Comment (vrabaud) [inline opencv_apps/CMakeLists.txt]: Better please use `if(TARGET opencv_optflow)`

**Comment by vrabaud** [comment]
You should also disable test-simple_flow.test

File: opencv_apps/CMakeLists.txt (hunk 1 above)
Comment (k-okada) [inline opencv_apps/CMakeLists.txt]: on indigo/opencv2 environment, we do not have opencv_optflow nor TARGET openecv_optflow

Comment (vrabaud) [inline opencv_apps/CMakeLists.txt]: oh right, then `if(OpenCV_VERSION VERSION_LESS "3.0" OR TARGET opencv_optflow)`

**Comment by trainman419** [comment]
Instead of calling the check to see if optflow exists multiple times, it's probably better to call that check once, and set a variable if optflow exists; then all of the other compilation steps that depend on optflow can use that variable.

```
if( ... however you decide to detect optflow ... )
     set(OPENCV_HAVE_OPTFLOW TRUE)
endif()

...

if(${OPENCV_HAVE_OPTFLOW})
   # build something that needs optflow
endif()

# repeat as needed for tests, etc
```

**Comment by vrabaud** [comment]
agreed with @trainman419 
- disable the test too. Thx.

**Comment by k-okada** [comment]
fixed, see latest commit

## 

◉ Kei Okada

File: opencv_apps/CMakeLists.txt [hunk 2]
```diff
@@ -70,6 +74,10 @@ catkin_package()
 include_directories( ${catkin_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS})
 
 ## Declare a cpp library
+if(${OPENCV_HAVE_OPTFLOW})
```
Comment (vrabaud) [inline opencv_apps/CMakeLists.txt]: The proper way to check for a variable is `if(OPENCV_HAVE_OPTFLOW)`. https://cmake.org/cmake/help/v3.0/command/if.html
`If(<variable>)` is a bit like what is done above with `OpenCV_VERSION`.

Comment (k-okada) [inline opencv_apps/CMakeLists.txt]: ok, fixed

## 

◉ Kei Okada

**Comment by vrabaud** [comment]
Thx a bunch !

**Pull request merged**